internal/client.go

```diff
@@ -538,7 +538,8 @@ type (
 		MaxPayloadSize int
 
 		// Advanced dial options for gRPC connections. These are applied after the internal default dial options are
-		// applied. Therefore any dial options here may override internal ones.
+		// applied. Therefore any dial options here may override internal ones. Dial options WithBlock, WithTimeout,
+		// WithReturnConnectionError, and FailOnNonTempDialError are ignored.
```

Comment:
```suggestion
		// WithReturnConnectionError, and FailOnNonTempDialError are ignored since [grpc.NewClient] is used
```

Might as well tell them why, but no biggie